pkg/services/rendering/rendering_test.go

```diff
@@ -0,0 +1,67 @@
+package rendering
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetUrl(t *testing.T) {
+	path := "render/d-solo/5SdHCadmz/panel-tests-graph?orgId=1&from=1587390211965&to=1587393811965&panelId=5&width=1000&height=500&tz=Europe%2FStockholm"
+	rs := &RenderingService{
+		Cfg: setting.NewCfg(),
+	}
+
+	t.Run("When renderer url configured", func(t *testing.T) {
+		rs.Cfg.RendererUrl = "http://localhost:8081/render"
+
+		t.Run("Should return path", func(t *testing.T) {
+			url := rs.getURL(path)
+			require.Equal(t, path+"&render=1", url)
+		})
```

Comment:
Basically, this case differs from the other one by not setting rs.Cfg.RendererCallbackUrl. Is this really a valid case? I'm not familiar with the semantics of the GetURL method, but I'm wondering why it should return a basically invalid URL (just starting with the path) if rs.Cfg.RendererCallbackUrl hasn't been set?

Reply:
Good catch. It's actually some logic in setting package setting callback URL:
https://github.com/grafana/grafana/blob/c26bfe9e03b92011bd402e0e64bb386817bca237/pkg/setting/setting.go#L926-L940

Will update the test